File: 3107-minimum-operations-to-make-median-of-array-equal-to-k/3107-minimum-operations-to-make-median-of-array-equal-to-k.cpp
class Solution {
public:
    long long minOperationsToMakeMedianK(vector<int>& nums, int k) {
        if(nums.size()==1) return abs(k-nums[0]);
        
        sort(nums.begin(), nums.end());
       
        long long ans=0;
        int n = nums.size();
       
        int idx = lower_bound(nums.begin(), nums.end(), k) - nums.begin();
      
        if(idx==n/2) return abs(nums[n/2]-k);
         
        if(idx<n/2){
             for(int i=idx;i<=n/2;i++){
                 ans+=abs(nums[i]-k);
             }
        }
        
        else if(idx>n/2){
             for(int i=n/2;i<idx;i++){
                 ans+=abs(nums[i]-k);
             }
        }

    return ans;
    }
};

File: 3107-minimum-operations-to-make-median-of-array-equal-to-k/3107-minimum-operations-to-make-median-of-array-equal-to-k.py
class Solution:
    def minOperationsToMakeMedianK(self, nums: List[int], k: int) -> int:

        if len(nums) == 1:
            return abs(k - nums[0])

        nums.sort()

        mid = len(nums) // 2

        if nums[mid] == k:
            return 0
        elif nums[mid] > k:
            counter = 0
            i = mid
            while nums[i] > k and i >= 0:
                counter += abs(k - nums[i])
                i -= 1
            return counter
        else:
            counter = 0
            i = mid
            while i < len(nums) and nums[i] < k:
                counter += k - nums[i]
                i += 1
            return counter